include/deal.II/matrix_free/fe_evaluation.h

```diff
@@ -5276,8 +5276,8 @@ namespace internal
   template <int dim, typename Number>
   struct EvaluatorTensorProduct<evaluate_general,dim,-1,0,Number>
   {
-    static const unsigned int dofs_per_cell = numbers::invalid_unsigned_int;
-    static const unsigned int n_q_points = numbers::invalid_unsigned_int;
+    static const unsigned int dofs_per_cell = 0;
+    static const unsigned int n_q_points = 0;
```

Comment:
Yea, I hear your pain. Would it work to write it as follows?
```
if (/*unsigned int */ temp_size < 100)
  {
    ptr1 = &temp[0];
    ptr2 = ptr1 + temp_size;
  }
else if (temp_size != invalid_unsigned_int)
  {
    ptr1 = somewhere_else;
    ptr2 = ptr1 + temp_size;
  }
else
  {
    ptr1 = somewhere_else;
    ptr2 = ptr1 + size_from_somewhere_else;
  }
```
I suspect you don't know whether that would work, because you can't test :-( But it may be clearer to read it that way, and if we get lucky, `temp_size` is constant propagated so that the `if` branches are all eliminated anyway.